tools/mkdocs_macros_bids/macros.py

```diff
@@ -9,6 +9,13 @@
 
 from examplecode import example
 
+def _get_source_path(level=1):
+    import inspect
+    caller = inspect.currentframe().f_back
+    for _ in range(level):
+        caller = caller.f_back
+    return caller.f_locals["_Context__self"]["page"].file.src_path
```

Comment:
could you write a docstr with a quick summary of what this does? to make it easier for future maintainers who may not be fully aware of what is happening here.

Reply:
Done in https://github.com/bids-standard/bids-specification/pull/1096/commits/1aec282e7f6d8838aecb26f1ab0ed432215b4b41.